xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -631,13 +631,16 @@ public void run() {
             }
             try {
                 Date cleanUpDate = getCleanupDate();
-                Iterator<LocalMUCRoom> it = localMUCRoomManager.getRooms().iterator();
-                while (it.hasNext()) {
-                    LocalMUCRoom room = it.next();
-                    Date emptyDate = room.getEmptyDate();
-                    if (emptyDate != null && emptyDate.before(cleanUpDate)) {
-                        removeChatRoom(room.getName());
-                    }
+                if (cleanUpDate!=null)
+                {
+	                Iterator<LocalMUCRoom> it = localMUCRoomManager.getRooms().iterator();
+	                while (it.hasNext()) {
+	                    LocalMUCRoom room = it.next();
+	                    Date emptyDate = room.getEmptyDate();
+	                    if (emptyDate != null && emptyDate.before(cleanUpDate)) {
+	                        removeChatRoom(room.getName());
+	                    }
+	                }
```

Comment:
Please make sure that you use the same level of indentation. You're switching from 4 space characters to 2, for some reason.